commit 324f9aab3dc9d87b85ec3eb25392907b85e9a3ed (refs/changes/19/719/2)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-10-19T00:46:45+00:00 (3 years ago)
    
    MANTA-3008 binder always serving TTLs as 5 seconds
    Reviewed by: Cody Peter Mello <cody.mello@joyent.com>

diff --git a/lib/server.js b/lib/server.js
index 6821828..0b7f444 100644
--- a/lib/server.js
+++ b/lib/server.js
@@ -162,7 +162,15 @@ function resolve(options, query, cb) {
                         query.setError('eserver');
                 } else {
                         var addr;
-                        var ttl = record.ttl;
+
+                        var ttl = (record[record.type] || {}).ttl;
+                        /*
+                         * Default the TTL to 30 seconds (the default ZK
+                         * session timeout)
+                         */
+                        if (ttl === undefined)
+                                ttl = 30;
+
                         if (service !== undefined &&
                             record.type !== 'service') {
                                 query.setError('eserver');
@@ -229,6 +237,7 @@ function resolve(options, query, cb) {
                                             ports.length < 1)
                                                 ports = [s.port];
                                         var ar, sr, nm;
+                                        var rttl = host[host.type].ttl;
                                         if (service !== undefined) {
                                                 nm = host.name + '.' + domain;
                                                 ports.forEach(function (p) {
@@ -240,11 +249,11 @@ function resolve(options, query, cb) {
                                                 });
                                                 ar = new ARecord(a);
                                                 query.addAdditional(nm, ar,
-                                                    ttl);
+                                                    rttl);
                                         } else {
                                                 ar = new ARecord(a);
                                                 query.addAnswer(domain, ar,
-                                                    ttl);
+                                                    rttl);
                                         }
                                 }
                                 break;
